src/Symfony/Component/Message/Handler/MessageHandlerCollection.php

```diff
@@ -37,8 +37,8 @@ public function __construct(array $handlers)
 
     public function __invoke($message)
     {
-        foreach ($this->handlers as $handler) {
-            yield $handler($message);
-        }
+        return array_map(function ($handler) use ($message) {
```

Comment:
@sroze: Sorry I had to revert this, otherwise the generator was returned from this and no handler was called.

Reply:
Oh... I see, good point! Good reason to revert it 👍

Reply:
FYI I've merged this change already in the main branch.